modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala

```diff
@@ -92,35 +90,51 @@ class SeqexecEngine(settings: SeqexecEngine.Settings) {
     type SequenceAR = engine.Sequence[engine.Action \/ engine.Result]
     type StepAR = engine.Step[engine.Action \/ engine.Result]
 
+    def viewSequence(seq: SequenceAR, st: SequenceState): SequenceView = {
+
+      def engineSteps(seq: SequenceAR): List[Step] = {
+
+        def viewStep(step: StepAR): Step =
+          StandardStep(
+            step.config,
+            engine.Step.status(step),
+            // TODO: Implement breakpoints at Engine level
+            breakpoint = false,
+            // TODO: Implement skipping at Engine level
+            skip = false,
+            configStatus = Map.empty,
+            // TODO: Implement standard step at Engine level
+            observeStatus = ActionStatus.Pending
+          )
 
-    def viewSequence(seq: SequenceAR, st: SequenceState): SequenceView =
+        // Couldn't find this on Scalaz
+        def splitWhere[A](l: List[A])(p: (A => Boolean)): (List[A], List[A]) =
+          l.splitAt(l.indexWhere(p))
+
+        seq.steps.map(viewStep) match {
+          // Find first Pending Step when no Step is Running and mark it as Running
+          case steps if st === SequenceState.Running && steps.toList.all(_.status =/= StepState.Running)=>
+            val (xs, (y :: ys)) = splitWhere(steps)(_.status === StepState.Pending)
+            xs ++ (
+              // TODO: Why it doesn't have `.copy`?
```

Comment:
I was very surprised, but it's not there. If you find the issue let me know because it doesn't make sense to me either.